### PR TITLE
feat: support private helm repos

### PR DESCRIPTION
**Pull request by TylerGillson**
- Add support for installing plugin charts within Helm repos using basic auth
- Fix previously unhandled race conditions when updating the ValidatorConfig object and its status subresource

### CODE REVIEW

File: internal/controller/validatorconfig_controller.go [hunk 1]
```diff
@@ -93,12 +103,17 @@ func (r *ValidatorConfigReconciler) Reconcile(ctx context.Context, req ctrl.Requ
...
-	r.Log.V(0).Info("Ensured ValidatorConfig finalizer")
+	// TODO: implement a proper patcher to avoid this hacky approach with retries & global vars
+	defer func() {
+		// Always update the ValidationConfig with a retry due to race condition with removeFinalizer.
+		for i := 0; i < constants.StatusUpdateRetries; i++ {
+			annotationErr := r.updateVc(ctx)
+			statusErr := r.updateVcStatus(ctx)
+			if annotationErr == nil && statusErr == nil {
+				break
+			}
```
Comment (ahmad-ibra) [inline internal/controller/validatorconfig_controller.go:114]: not sure what the nature of this race condition is, but would it make sense to add a small timeout between retries? or are there enough retries that the race condition is going to be resolved before we run out of retries?

File: internal/controller/validatorconfig_controller.go [hunk 2]
```diff
@@ -116,18 +131,49 @@ func (r *ValidatorConfigReconciler) SetupWithManager(mgr ctrl.Manager) error {
...
+	}
+
+	// ensure cleanup finalizer
+	ensureFinalizer(ctx, r.Client, vc, CleanupFinalizer)
+	r.Log.V(0).Info("Ensured ValidatorConfig finalizer")
+
+	vc.Annotations = annotations
+
+	if err := r.Client.Update(ctx, vc); err != nil {
+		r.Log.V(1).Info("warning: failed to update ValidatorConfig annotations", "error", err)
```
Comment (ahmad-ibra) [inline internal/controller/validatorconfig_controller.go:148]: curious why this isn't a case where you'd want to return an error and retry

File: pkg/helm/helm.go [hunk 3]
```diff
@@ -59,7 +59,23 @@ func (c *helmClient) exec(args []string) error {
 		return nil
 	}
 
-	fmt.Println("helm " + strings.Join(args, " "))
+	sb := strings.Builder{}
+	mask := false
+	for _, a := range args {
+		if mask {
+			sb.WriteString("***** ")
```
Comment (ahmad-ibra) [inline pkg/helm/helm.go:66]: nice improvement 🙂

**Review by ahmad-ibra — DISMISSED** [review]
Left a couple questions but LGTM

File: internal/controller/validatorconfig_controller.go (hunk 1 above)
Comment (TylerGillson) [reply internal/controller/validatorconfig_controller.go]: resolved in 4bc01cfb93a3970e4144cef2c7c03f2424494ff7

File: internal/controller/validatorconfig_controller.go (hunk 2 above)
Comment (TylerGillson) [reply internal/controller/validatorconfig_controller.go]: resolved in 9d352385eed91878e6e463b08b54d6a79e5fc20d

**Review by ahmad-ibra — APPROVED** [review]

**Pull request merged**